File: 1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.cpp
class Solution {
public:
    bool canBeEqual(vector<int>& target, vector<int>& arr) {
        map<int, int> mp;
        for (int x : target) {
            mp[x]++;
        }
        for (int x : arr) {
            if (mp.find(x) == mp.end()) {
                return false;
            }
            mp[x]--;
            if (mp[x] == 0) {
                mp.erase(x);
            }
        }
        return mp.size() == 0;
    }
};





// class Solution {
// public:
//     bool canBeEqual(vector<int>& target, vector<int>& arr) {
//         sort(target.begin(), target.end());
//         sort(arr.begin(), arr.end());
//         return arr == target ? true : false;
//     }
// };

File: LeetCode/1460-make-two-arrays-equal-by-reversing-subarrays/1460-make-two-arrays-equal-by-reversing-subarrays.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
        HashMap<Integer, Integer> mp = new HashMap<>();
        for (int x : target) {
            mp.put(x, mp.getOrDefault(x, 0) + 1);
        }
        for (int x : arr) {
            if (!mp.containsKey(x)) {
                return false;
            }
            mp.put(x, mp.getOrDefault(x, 0) - 1);
            if (mp.get(x) == 0) {
                mp.remove(x);
            }
        }
        return mp.size() == 0;
    }
}




// class Solution {
//     public boolean canBeEqual(int[] target, int[] arr) {
//         Arrays.sort(target);
//         Arrays.sort(arr);
//         return Arrays.equals(target, arr);
//     }
// }